transport.go

```diff
@@ -68,15 +67,14 @@ func NewTransport(c net.Conn, l LogFactory, wef WrapErrorFunc) Transporter {
 		cdec:             cdec,
 		log:              log,
 		wrapError:        wef,
-		encodeCh:         make(chan []byte),
-		encodeResultCh:   make(chan error),
+		encodeCh:         make(chan encodingBundle, 1),
```

Comment:
Right, I just noticed that. I think that it should, as otherwise the return value is meaningless. I noticed a similar issue with handleCall -- I'll write up comments on those.